### PR TITLE
Add e2e tests and commands for auth flows on devel

### PR DESCRIPTION
**Pull request by patcon**
Closes https://github.com/GCTC-NTGC/gc-digital-talent/issues/2571

Rebased off of #2813, so blocked on that for now

## To Do
- [x] updated e2e test workflow (lock node/npm version, add /etc/hosts entry)
- [x] ~~added affordances for cross-origin tests if needed (`api/config/sessions.php`, `frontend/.apache_env`, comment about `cypress.json`~~ `reverted`
  - [x] ~~document this better (README)~~
- [x] added `auth.spec.js` tests (cookies, localstorage tokens, login/logout commands)
  - [x] remove unused helper functions from spec
- [x] document writing e2e tests
- [x] install and implement [`testing-library/cypress`](https://testing-library.com/docs/cypress-testing-library/intro/) to allow usage of familiar testing-library selectors
  - [x] document
- [x] added [support functions for working with graphQL](https://docs.cypress.io/guides/testing-strategies/working-with-graphql)
  - [x] document
- [x] ~~improve "restricted page" test to confirm no data returns from graphql~~ `wontdo`
  - https://talent-cloud.slack.com/archives/CB424V61J/p1653058682633159
- [x] rebased on mock auth server branch and mainline changes

### CODE REVIEW

**Comment by patcon** [comment]
Ah, changes to admin dashboard changed this. Fixing before opening for review

**Comment by patcon** [comment]
Crossing my fingers that this is ready for review tomorrow.

Also, before anyone reviews, I'd like to write a bullet point summary of what's going on, as the commit log isn't as clean as I'd like.

**Comment by patcon** [comment]
k this is rebased, and all the tests are enabled. CI should pass, and then it's ready for code review (assuming #2813 is stable)

File: .gitignore [hunk 1]
```diff
@@ -3,10 +3,11 @@ node_modules
 frontend/*/package-lock.json
 package-lock.json
 
+# TODO: Remove once all feature branches with auth app are complete.
```
Comment (patcon) [inline .gitignore:6]:
```suggestion
```

**Review by petertgiles — APPROVED** [review]
Wow, super impressive.  I feel like I've just taken my first step on a cypress journey even though most of it went over my head. :smile:

**Review by esizer — APPROVED** [review]
Very impressive! 👍  I'd be lying if I said I understood everything going on here but I tested and it all works!

File: docker-compose.yml [hunk 2]
```diff
@@ -82,7 +82,13 @@ services:
       #   login form will be used.
       # - we match specifically for client_id `fake-development-client-key`,
       #   which must be set in `.env`.
-      JSON_CONFIG_PATH: /app/conf/mock-auth-server.json
+      # - for now, we set interactiveLogin:false, as we can assume that any
```
[the remark this answers is not in the document]
Comment (patcon) [reply docker-compose.yml:85]: Ah good catch! Thanks peter

File: docker-compose.yml [hunk 3]
```diff
@@ -82,7 +82,13 @@ services:
...
+      #   login is for the ADMIN user with `sub` of `admin@test.com`. When we
+      #   need to login with multiple users, we'll need to enabled
+      #   interativeLogin and un-hardcode the "sub" in the response.
```
Comment (patcon) [inline docker-compose.yml:88]:
```suggestion
      # - we set interactiveLogin:true, for easy selection of any test user
      #   (for which subject/sub will be something like `admin@test.com`).
```

**Comment by patcon** [comment]
Ok, going to merge as soon as this branch update passes 🤞

**Pull request merged**